datahub/opal.py

```diff
@@ -109,14 +61,18 @@ def __init__(self, pandas_obj: pd.DataFrame) -> None:
         """
         self._obj = pandas_obj
 
-    def append(self, data: dict[str, float]) -> None:
+    def append(self, data: dict[str, Any]) -> None:  # type: ignore[misc]
         """Function to append new data to existing dataframe.
 
         Args:
             data: The raw opal data posted to the API
         """
         row = get_opal_row(data)
-        self._obj.loc[data["frame"]] = row  # type: ignore[call-overload]
+        if "array" in data:
```

Comment:
I would prefer that this append function take either a dictionary or a list as input and this check is more like:
```suggestion
        if isinstance(data, list):
```